python_modules/dagster/dagster/_core/definitions/freshness_checks/last_update.py

```diff
@@ -0,0 +1,63 @@
+import datetime
+from typing import Optional, Sequence, Union
+
+from dagster._annotations import experimental
+from dagster._core.definitions.asset_check_spec import AssetCheckSeverity
+
+from ..asset_checks import AssetChecksDefinition
+from ..assets import AssetsDefinition, SourceAsset
+from ..events import CoercibleToAssetKey
+from .utils import (
+    DEFAULT_FRESHNESS_CRON_TIMEZONE,
+    DEFAULT_FRESHNESS_SEVERITY,
+    build_freshness_checks_for_assets,
+)
+
+
+@experimental
+def build_last_update_freshness_checks(
+    *,
+    assets: Sequence[Union[CoercibleToAssetKey, AssetsDefinition, SourceAsset]],
+    lower_bound_delta: datetime.timedelta,
+    freshness_cron: Optional[str] = None,
+    freshness_cron_timezone: str = DEFAULT_FRESHNESS_CRON_TIMEZONE,
+    severity: AssetCheckSeverity = DEFAULT_FRESHNESS_SEVERITY,
+) -> Sequence[AssetChecksDefinition]:
+    r"""For each provided asset, constructs a freshness check definition.
+
+    This check passes if the asset is considered "fresh" by the time execution begins. An asset is
+    considered fresh if a record exists with an update timestamp greater than the "lower bound"
+    derived from the freshness_cron and lower_bound_delta parameters. The "lower bound" is defined
```

Comment:
`update_window_duration` seems solid to me.

`expiration_cron` feels off.  Like an expiration date on a package of food, it makes me think that, as soon as the asset hits the expiration date, it's no longer considered fresh.

I think the way most users will want to think about this parameter is as a deadline.

Reply:
Yeah def not set on expiration.

My thinking here is that if we use the "window" language we need a variable name that makes it unambiguous that it is the cron that determines the _end_ rather than the _start_ of the window. Deadline does that too IMO.

Reply:
> My thinking here is that if we use the "window" language we need a variable name that makes it unambiguous that it is the cron that determines the end rather than the start of the window

Got it - that makes sense